### DISCUSSION TITLE
Writing tests

### DISCUSSION THREAD

**Discussion by bon** [coalton-lang/coalton#373] [Help!]
What's the wisdom on how to write tests using fiasco? 

To use the symbolic differentiation example on the Coalton github home page, if you add the appropriate `define-instance` for equality you can do this at the command line
```
(coalton (== (dt (E+ (EVar (Symbol "t"))
                                (EConst 1)))
                    (E+ (EConst 1) (EConst 10))))
```
Now how to write that as a test? Variations on the theme below don't seem to work:
```
(deftest diff-test ()
  (is (coalton (== (dt (E+ (EVar (Symbol "t"))
                                       (EConst 1)))
                           (E+ (EConst 1) (EConst 10))))))
```

**Comment by stylewarning** [comment]
What doesn't work about that example? Are you sure you're in the right package? (I admit I haven't actually tried to run it... *yet*.)

**Reply by bon** [reply]
Something like this... https://gist.github.com/bon/59c0feb3546a9d3de1d466ae201bf8db
